src/test/amount_tests.cpp

```diff
@@ -86,6 +90,31 @@ BOOST_AUTO_TEST_CASE(GetFeeTest)
     CFeeRate(MAX_MONEY, std::numeric_limits<size_t>::max() >> 1).GetFeePerK();
 }
 
+BOOST_AUTO_TEST_CASE(CFeeRateConstructorWithUnitsTest)
+{
+    // Test CFeerate(CAmount fee_rate, FeeEstimatemode mode) constructor
+    // ...with BTC/kvB, returns same values as CFeeRate(CAmount fee_rate) ctor
+    BOOST_CHECK(CFeeRate(CAmount(-1), FeeEstimateMode::BTC_KVB) == CFeeRate(-1));
+    BOOST_CHECK(CFeeRate(CAmount(0), FeeEstimateMode::BTC_KVB) == CFeeRate(0));
+    BOOST_CHECK(CFeeRate(CAmount(1), FeeEstimateMode::BTC_KVB) == CFeeRate(1));
+    BOOST_CHECK(CFeeRate(CAmount(26), FeeEstimateMode::BTC_KVB) == CFeeRate(26));
+    BOOST_CHECK(CFeeRate(CAmount(123), FeeEstimateMode::BTC_KVB) == CFeeRate(123));
+    // ...with sat/vB, returns values that are 1e5 smaller
+    BOOST_CHECK(CFeeRate(CAmount(-100000), FeeEstimateMode::SAT_VB) == CFeeRate(-1));
+    BOOST_CHECK(CFeeRate(CAmount(-99999), FeeEstimateMode::SAT_VB) == CFeeRate(0));
+    BOOST_CHECK(CFeeRate(CAmount(0), FeeEstimateMode::SAT_VB) == CFeeRate(0));
+    BOOST_CHECK(CFeeRate(CAmount(99999), FeeEstimateMode::SAT_VB) == CFeeRate(0));
+    BOOST_CHECK(CFeeRate(CAmount(100000), FeeEstimateMode::SAT_VB) == CFeeRate(1));
+    BOOST_CHECK(CFeeRate(CAmount(100001), FeeEstimateMode::SAT_VB) == CFeeRate(1));
+    BOOST_CHECK(CFeeRate(CAmount(2690000), FeeEstimateMode::SAT_VB) == CFeeRate(26));
+    BOOST_CHECK(CFeeRate(CAmount(123456789), FeeEstimateMode::SAT_VB) == CFeeRate(1234));
+    // ...with invalid unit values returns CFeeRate(0)
+    BOOST_CHECK(CFeeRate(CAmount{123}, FeeEstimateMode::UNSET) == CFeeRate{0});
+    BOOST_CHECK(CFeeRate(CAmount{123}, std::numeric_limits<size_t>::max()) == CFeeRate{0});
+    BOOST_CHECK(CFeeRate(CAmount{123}, 0) == CFeeRate{0});
+    BOOST_CHECK(CFeeRate(CAmount{123}, -1) == CFeeRate{0});
```

Comment:
style nit in 4cccf2d88377a9f813ab6d63889cbb5136c31aad

Might be good to make this impossible to compile instead of a runtime-just-return-0. Can be achieved via `constexpr if` (and some template magic) or by having a `CFeeRate::FromSatB(amount)` and `FromBtcKb(amount)` helper.

Reply:
Oh oops hm, these lines 113-115 aren't testing the new constructor, but the `CFeeRate(CAmount, size_t)` one, so it was mistake to add them here. Will move them into the tests for that constructor and see if anything needs to be done there; normally the ctor should just set `nSatoshisPerK` to 0 in that case.

Reply:
(my comment should be on the `BOOST_CHECK(CFeeRate(CAmount{123}, FeeEstimateMode::UNSET) == CFeeRate{0})` line)

Reply:
Good idea, much better...switching to named constructors.

Reply:
done, and a leaner implementation, thanks